addons/mail/static/src/messaging/component/composer_text_input/composer_text_input.js

```diff
@@ -144,10 +183,83 @@ class ComposerTextInput extends Component {
     }
 
+    /**
+     * Validates user's current typing as a correct mention keyword
+     * in order to trigger mentions suggestions display.
+     * Returns the mention keyword without the delimiter if it has been validated
+     * and false if not
+     *
+     * @private
+     * @param delimiter
+     * @param beginningOnly
+     * @returns {string|boolean}
+     */
+    _validateMentionKeyword(delimiter, beginningOnly) {
+        const leftString = this.getContent().substring(0, this._getSelectionStart());
+
+        // use position before delimiter because there should be whitespaces
+        // or line feed/carriage return before the delimiter
+        const beforeDelimiterPosition = leftString.lastIndexOf(delimiter) - 1;
+        if (beginningOnly && beforeDelimiterPosition > 0) {
+            return false;
+        }
+        let searchStr = this.getContent().substring(beforeDelimiterPosition, this._getSelectionStart());
+        // regex string start with delimiter or whitespace then delimiter
+        const pattern = "^"+delimiter+"|^\\s"+delimiter;
+        const regexStart = new RegExp(pattern, 'g');
+        // trim any left whitespaces or the left line feed/ carriage return
+        // at the beginning of the string
+        searchStr = searchStr.replace(/^\s\s*|^[\n\r]/g, '');
+        if (regexStart.test(searchStr) && searchStr.length) {
+            searchStr = searchStr.replace(pattern, '');
+            return !searchStr.includes(' ') && !/[\r\n]/.test(searchStr)
+                ? searchStr.replace(delimiter, '')
+                : false;
+        }
+        return false;
+    }
+
+    /**
+     * Detects if mentioned partners are still in the composer text input
+     * and removes them if not.
+     *
+     * @private
+     */
+    _validatePartnerIds() {
+        const inputMentions = this.getContent().match(new RegExp("@"+'[^ ]+(?= |&nbsp;|$)', 'g'));
+        if (inputMentions) {
+            for (const mention of this.composer.mentionedPartners) {
+                let inputMention = inputMentions.find(item => {
+                    return item === ("@" + mention.name).replace(/ /g, '\u00a0');
+                });
+                if (!inputMention) {
+                    this.composer.update({ mentionedPartners: [['link', mention]] });
+                }
+            }
+        }
+        else {
+            this.composer.update({ mentionedPartners: [['unlink-all']] });
+        }
+    }
+
     //--------------------------------------------------------------------------
     // Handlers
     //--------------------------------------------------------------------------
 
+    /**
+     * @private
+     * @param ev
+     */
+    _onClickMentionItem(ev) {
+        ev.preventDefault();
+        this.composer.update({
+            activePartner: [['link', ev.detail.partner]],
+        });
+        this.composer.insertSuggestedPartner(this._mentionKeyword);
+        this._update();
+        this.focus();
```

Comment:
This focus should not be here I think, related to the other task about focus, but it should be linked to entity attr and updated in mounted/patched accordingly.